Chapter1/1_Is Unique/IsUnique.py

```diff
@@ -1,13 +1,23 @@
 # O(N)
 import unittest
-
+import numpy as np
+import sys
 
 def unique(string):
     # Assuming character set is ASCII (128 characters)
-    if len(string) > 128:
+    #max_len = 128
+    
+    # Assuming all unicode set
+    max_len = sys.maxunicode
+    
+    if len(string) > max_len:
         return False
-
-    char_set = [False for _ in range(128)]
+        
+        
+    # np.ones or zeros is much faster than list comprehension 
```

Comment:
Using numpy is good for additional questions after formulating the algo, but I would argue that it shouldn't be required to know numpy or even utilize it "out of the box". It is a data science library and this algo is simple and should be solvable using most standard/built in libraries. 

You probably don't even need list comprehension here.

Reply:
wow, didn't expect that I get the review. 
Anyway, so for your question, I think it depends on the purpose of using python for this kind of coding test. 
The purpose of this stack is to use python to solve those coding tests for interviews, and it is also a good exercise for those who just wants to learn python by solving problems.  

For job interviews, if I am the guy to hire a python developer, and I see someone solving the problem without using list comprehension where it should be used for performance gain, I will really doubt if he knows python well enough for work. 

Numpy is kind of a must know for Python developers who wants to write efficient code. I agree that, it might be overkill in this case, Just to get optimal solution.